decode.go

```diff
@@ -1442,7 +1442,7 @@ func (d *Decoder) decodeStruct(ctx context.Context, dst reflect.Value, src ast.N
 	}
 
 	if d.validator != nil {
-		if err := d.validator.Struct(dst.Interface()); err != nil {
+		if err = d.validator.Struct(dst.Addr().Interface()); err != nil {
```

Comment:
dst is guaranteed to be addressable. See [this line](https://github.com/goccy/go-yaml/blob/b6dea53c6b11947b106c90b66c1b028a0d304b91/decode.go#L949).